### PR TITLE
fix: remove 08-wasm from 02-client exported

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/testing/wasm_endpoint.go [hunk 1]
```diff
@@ -50,3 +50,12 @@ func (endpoint *WasmEndpoint) CreateClient() error {
 
 	return nil
 }
+
+// AllowWasmClients adds 08-wasm to the list of allowed clients
+func AllowWasmClients(chain *ibctesting.TestChain) {
```
Comment (crodriguezvega) [inline modules/light-clients/08-wasm/testing/wasm_endpoint.go:55]: I can add this to a different file, if folks prefer that.

File: e2e/tests/wasm/grandpa_test.go
Comment (damiannolan) [inline e2e/tests/wasm/grandpa_test.go:1]: Could we just hardcode `08-wasm` into allowed clients at genesis? I feel like there's already a many proposals run in these tests. Each one uses a store code proposal, and the client expiry/recovery test uses both store code and recover client props.

File: modules/light-clients/08-wasm/types/keys.go [hunk 2]
```diff
@@ -6,4 +6,7 @@ const (
 
 	// StoreKey is the store key string for 08-wasm
 	StoreKey = ModuleName
+
+	// Wasm is used to indicate that the light client is a on-chain wasm program
+	Wasm string = ModuleName
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/types/keys.go:11]: do we need the `string` beside the const name? 

suggestion for godoc:

```suggestion
	// Wasm is the client type for IBC light clients created using 08-wasm
	Wasm string = ModuleName
```

**Review by damiannolan — COMMENTED** [review]
I think it would be good if we could avoid running another gov proposal in e2e.

File: e2e/tests/wasm/grandpa_test.go
Comment (crodriguezvega) [reply e2e/tests/wasm/grandpa_test.go:1]: It's a good idea. I can look into it, but I'd rather tackle that in a separate PR, if you're ok with that.

Comment (damiannolan) [reply e2e/tests/wasm/grandpa_test.go:1]: https://github.com/cosmos/ibc-go/issues/5311

**Review by damiannolan — APPROVED** [review]

**Pull request merged**